### PR TITLE
Add API change warning plots

### LINKED ISSUE #477: Add warning that `plot_*` is going to change
Related to #465 

Inform the user about upcoming API change for `plot_*()` functions.

### PR DESCRIPTION
**Pull request by MonikaFu**
Closes #477 

In this PR I add warnings that inform the used about upcoming API change.

### CODE REVIEW

File: R/plot_emission_intensity.R [hunk 1]
```diff
@@ -30,6 +30,15 @@
 plot_emission_intensity <- function(data,
                                     span_5yr = FALSE,
                                     convert_label = identity) {
+  lifecycle::deprecate_soft(
+      when = "0.4.0",
+      what = "plot_emission_intensity(data = 'must be prepped already')",
+      details = c(
+        glue("From the next release the call to the plot function will change to:
+             data %>% r2dii.plot::prep_emission_intensity() %>% r2dii.plot::plot_emission_intensity()."),
+        "Custom data preparation will also become possible."
```
Comment (jdhoffa) [inline R/plot_emission_intensity.R:39]: Maybe write the warning without knowledge of `dplyr` or pipes? Perhaps something like: 
"From the next release, please call: `prep_emission_intensity(data)` prior to calling `plot_emission_intensity()`"

Also I don't know if you need to namespace the package here, it seems implied that this deprecation is in relation to `r2dii.plot`?

Comment (jdhoffa) [reply R/plot_emission_intensity.R:39]: I might include @maurolepore here, I don't have a lot of experience at all in this deprecation process, or best practices there.

Comment (MonikaFu) [reply R/plot_emission_intensity.R:39]: thanks @jdhoffa ! The `r2dii.plot` reference was a suggestion from @AlexAxthelm . The exclusion of `dplyr` seems fair. I will change that.

Comment (jdhoffa) [reply R/plot_emission_intensity.R:39]: !! ok cool. then i will also namespace my warnings in the other `r2dii` deprecations 8-)

File: R/utils.R [hunk 2]
```diff
@@ -280,3 +280,11 @@ integer_breaks <- function(n = 5, ...) {
...
   return(fxn)
 }
+
+api_warning_details <- function(function_name) {
+  c(
+    glue("From the next release you will need to call `r2dii.plot::{function_name}(data)`
+             prior to calling `r2dii.plot::{function_name}()`."),
+    "Alternatively custom data preparation will also become possible."
+  )
+}
```
Comment (MonikaFu) [inline R/utils.R:290]: @jdhoffa please see the new details of the warning message. I hope you agree with these.

File: R/utils.R [hunk 3]
```diff
@@ -280,3 +280,11 @@ integer_breaks <- function(n = 5, ...) {
   }
   return(fxn)
 }
+
+api_warning_details <- function(function_name) {
+  c(
+    glue("From the next release you will need to call `r2dii.plot::{function_name}(data)`
+             prior to calling `r2dii.plot::{function_name}()`."),
+    "Alternatively custom data preparation will also become possible."
```
Comment (MonikaFu) [inline R/utils.R:288]: @jdhoffa please see the new details of the warning message. I hope you agree with these.

Comment (jdhoffa) [reply R/utils.R:288]: looks great :-)

**Review by jdhoffa — APPROVED** [review]
lgtm

**Pull request merged**